packages/fx-core/src/common/local/taskDefinition.ts

```diff
@@ -0,0 +1,242 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+"use strict";
+
+import { FolderName, ProgrammingLanguage, npmInstallCommand } from "./constants";
+import path from "path";
+import { isWindows } from "../deps-checker/util/system";
+
+export interface ITaskDefinition {
+  name: string;
+  command: string;
+  cwd: string | undefined;
+  isBackground: boolean;
+  isShell: boolean;
+  isCmd?: boolean;
+  args?: string[];
+  env?: { [key: string]: string };
+}
+
+export class CustomTaskDefinition {
+  static command(isWatchTask: boolean): string {
+    return isWatchTask ? "npm run watch:teamsfx" : "npm run dev:teamsfx";
+  }
+
+  static nameSuffix(isWatchTask: boolean): string {
+    return isWatchTask ? "watch" : "dev";
+  }
+
+  static frontend(workspaceFolder: string, isWatchTask: boolean): ITaskDefinition {
+    return {
+      name: `frontend ${CustomTaskDefinition.nameSuffix(isWatchTask)}`,
+      command: CustomTaskDefinition.command(isWatchTask),
+      cwd: path.join(workspaceFolder, FolderName.Frontend),
+      isShell: true,
+      isBackground: true,
+    };
+  }
+
+  static backend(
+    workspaceFolder: string,
+    isWatchTask: boolean,
+    funcBinFolders: string[] | undefined
+  ): ITaskDefinition {
+    return {
+      name: `backend ${CustomTaskDefinition.nameSuffix(isWatchTask)}`,
+      command: CustomTaskDefinition.command(isWatchTask),
+      cwd: path.join(workspaceFolder, FolderName.Function),
+      isShell: true,
+      isBackground: true,
+      env: funcBinFolders
+        ? {
+            // put portable func at the end since func checker prefers global func
+            PATH: `${process.env.PATH ?? ""}${path.delimiter}${funcBinFolders.join(
+              path.delimiter
+            )}`,
+          }
+        : undefined,
+    };
+  }
+
+  static bot(workspaceFolder: string, isWatchTask: boolean): ITaskDefinition {
+    return {
+      name: `bot ${CustomTaskDefinition.nameSuffix(isWatchTask)}`,
+      command: CustomTaskDefinition.command(isWatchTask),
+      cwd: path.join(workspaceFolder, FolderName.Bot),
+      isShell: true,
+      isBackground: true,
+    };
+  }
+}
+
+export class TaskDefinition {
+  static frontend(workspaceFolder: string): ITaskDefinition {
```

Comment:
Can we name it more specific for readability? e.g. `frontendStart` or `frontendStartTask`